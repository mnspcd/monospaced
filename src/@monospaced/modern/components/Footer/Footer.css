.mdrn .Footer {
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
  margin: 0 auto;
  max-width: 56.75em;
  padding-bottom: 1.5em;
  padding-left: 1em;
  padding-right: 1em;
  padding-top: 1.5em;
  width: 100%;
}

@media (--larger-viewport) {
  .mdrn .Footer {
    align-items: center;
    display: flex;
    height: 4.5em;
    justify-content: space-between;
    padding-bottom: 0;
    padding-top: 0;
  }
}

.mdrn .Footer-copyright {
  color: var(--color-blue-9);
  font-size: calc(1em * var(--type-scale-s));
  margin-bottom: calc(12rem / 16);
}

@media (--larger-viewport) {
  .mdrn .Footer-copyright {
    margin-bottom: 0;
  }
}

.mdrn .Footer-list {
  font-size: calc(1em * var(--type-scale-s));
  list-style: none;
  margin-bottom: 0;
  margin-top: 0;
  padding-left: 0;
}

.mdrn .Footer-listItem {
  display: inline-block;
  margin-right: 1em;
}

@media (--larger-viewport) {
  .mdrn .Footer-listItem {
    margin-right: 1.5em;
  }
}

.mdrn .Footer-listItem:last-child {
  margin-right: 0;
}

.mdrn .Footer-linkActive {
  color: var(--color-blue-8);
}
