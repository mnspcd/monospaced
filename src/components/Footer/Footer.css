.Footer {
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
  margin: 0 auto;
  max-width: 100%;
  padding-bottom: 1.5em;
  padding-left: 1em;
  padding-right: 1em;
  padding-top: 1.5em;
  width: 56.75em;
}

@media (--larger-viewport) {
  .Footer {
    align-items: center;
    display: flex;
    height: 4.5em;
    justify-content: space-between;
    padding-bottom: 0;
    padding-top: 0;
  }
}

.Footer-copyright {
  color: var(--mnspcd-color-blue-9);
  font-size: calc(1em * var(--type-scale-s));
  margin-bottom: calc(12rem / 16);
}

@media (--larger-viewport) {
  .Footer-copyright {
    margin-bottom: 0;
  }
}

.Footer-list {
  font-size: calc(1em * var(--type-scale-s));
  list-style: none;
  margin-bottom: 0;
  margin-top: 0;
  padding-left: 0;
}

.Footer-listItem {
  display: inline-block;
  margin-right: 1em;
}

@media (--larger-viewport) {
  .Footer-listItem {
    margin-right: 1.5em;
  }
}

.Footer-listItem:last-child {
  margin-right: 0;
}

.Footer-linkActive {
  color: var(--mnspcd-color-blue-8);
}
