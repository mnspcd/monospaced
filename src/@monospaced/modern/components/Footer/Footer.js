import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router";

import "./Footer.css";

/**
 * Use `Footer` to display copyright and supplemental links at the bottom of
 * a page.
 */
const Footer = ({ copyright, links = null, routes = null }) => {
  const path =
    routes && routes[routes.length - 1] && routes[routes.length - 1].path;

  return (
    <div className="Footer">
      <div className="Footer-copyright">
        Â© {copyright} 2024 {/* new Date().getFullYear() */}
      </div>
      {links && (
        <ul className="Footer-list">
          {links.map(link => {
            const { href, routerLink, text } = link;

            return (
              <li className="Footer-listItem" key={text}>
                {routerLink ? (
                  <Link
                    className={classNames({
                      [`Footer-link`]: true,
                      [`Footer-linkActive`]: href.replace(/\//g, "") === path,
                    })}
                    to={href}
                  >
                    {text}
                  </Link>
                ) : (
                  <a className="Footer-link" href={href}>
                    {text}
                  </a>
                )}
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
};

Footer.propTypes = {
  /**
   * Text to display in the Footer copyright declaration. The component provides
   * the copyright symbol and current year.
   */
  copyright: PropTypes.string.isRequired,

  /**
   * Supplemental links to display in the Footer.
   */
  links: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string.isRequired,
      routerLink: PropTypes.bool,
      text: PropTypes.string.isRequired,
    }),
  ),

  /**
   * React Router routes array, from which the Footer can extract the
   * current path.
   */
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
    }),
  ),
};

export default Footer;
