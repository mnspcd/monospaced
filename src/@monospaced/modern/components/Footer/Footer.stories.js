import { number, text } from "@storybook/addon-knobs";
import React from "react";

import Footer from "./Footer";

export default {
  component: Footer,
  title: "Components/Footer",
};

export const basic = () => (
  <Footer
    copyright="Monospaced"
    links={[
      {
        href: "https://example.com",
        text: "item-1",
      },
      {
        href: "https://example.com",
        text: "item-2",
      },
      {
        href: "https://example.com",
        text: "item-3",
      },
    ]}
  />
);

export const knobs = () => {
  const numberOfItems = number(
    "number of links",
    3,
    {
      max: 7,
      min: 1,
      range: true,
    },
    "Story",
  );
  const copyright = text("copyright", "Monospaced", "Footer");

  return (
    <Footer
      copyright={copyright}
      links={[...Array(numberOfItems).keys()].map(i => ({
        href: text("link href", "https://example.com", "Footer"),
        text: `${text("link text", "item", "Footer")}-${i + 1}`,
      }))}
    />
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
