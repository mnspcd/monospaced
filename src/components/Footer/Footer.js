import { Link } from "react-router";
import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import "./Footer.css";

const Footer = ({ copyright, links, routes }) => {
  const path = routes && routes[routes.length - 1].path;

  return (
    <div className="Footer">
      <div className="Footer-copyright">
        Â© {copyright} {new Date().getFullYear()}
      </div>
      {links && (
        <ul className="Footer-list">
          {links.map(link => {
            const { href, routerLink, text } = link;

            return (
              <li className="Footer-listItem" key={text}>
                {routerLink ? (
                  <Link
                    className={classNames({
                      [`Footer-link`]: true,
                      [`Footer-linkActive`]: href.replace(/\//g, "") === path,
                    })}
                    to={href}
                  >
                    {text}
                  </Link>
                ) : (
                  <a className="Footer-link" href={href}>
                    {text}
                  </a>
                )}
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
};

Footer.propTypes = {
  copyright: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string.isRequired,
      routerLink: PropTypes.bool,
      text: PropTypes.string.isRequired,
    }),
  ),
  routes: PropTypes.array,
};

export default Footer;
