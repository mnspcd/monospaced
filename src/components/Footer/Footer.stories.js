import React from "react";
import { number, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Footer from "./Footer";

storiesOf("Components/Footer", module).add("Component", () => {
  const numberOfItems = number(
    "number of links",
    3,
    {
      range: true,
      min: 1,
      max: 7,
    },
    "Story",
  );
  const copyright = text("copyright", "Monospaced", "Footer");

  return (
    <Footer
      copyright={copyright}
      links={[...Array(numberOfItems).keys()].map(i => ({
        href: text("link href", "https://example.com", "Footer"),
        text: `${text("link text", "item", "Footer")}-${i + 1}`,
      }))}
    />
  );
});
