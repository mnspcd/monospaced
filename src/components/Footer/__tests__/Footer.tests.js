import React from "react";
import { render } from "enzyme";

import Footer from "../Footer";

describe("Footer component", () => {
  it("should render correctly", () => {
    const component = render(<Footer copyright="copyright" />);
    expect(component).toMatchSnapshot();
  });

  it("should render links correctly", () => {
    const component = render(
      <Footer
        copyright="copyright"
        links={[
          {
            href: "href",
            routerLink: true,
            text: "text",
          },
          {
            href: "href",
            text: "text",
          },
        ]}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
