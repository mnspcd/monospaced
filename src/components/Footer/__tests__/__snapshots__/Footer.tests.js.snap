// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
<div
  class="Footer"
>
  <div
    class="Footer-copyright"
  >
    © copyright 2018
  </div>
</div>
`;

exports[`Footer component should render links correctly 1`] = `
<div
  class="Footer"
>
  <div
    class="Footer-copyright"
  >
    © copyright 2018
  </div>
  <ul
    class="Footer-list"
  >
    <li
      class="Footer-listItem"
    >
      <a
        class="Footer-link"
      >
        text
      </a>
    </li>
    <li
      class="Footer-listItem"
    >
      <a
        class="Footer-link"
        href="href"
      >
        text
      </a>
    </li>
  </ul>
</div>
`;
